packages/slice-machine/lib/models/common/widgets/Image/components.tsx

```diff
@@ -75,84 +78,103 @@ export const AddThumbnailButton = ({ onClick }) => (
+}> = ({ prefix, required = false, display }) => {
   if (!display) {
     return null;
   }
   const requiredChar = required ? "*" : "";
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-unsafe-argument
   const [field, meta, helpers] = useField(prefix);
 
   const createSetField =
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-
-
-      (key, fn = (v) => v) =>
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-      (e) => {
-        helpers.setTouched(true);
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-        const cast = fn(e.target.value);
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-        const value =
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-argument
-          e.target.type === "number" ? (isNaN(cast) ? null : cast) : cast;
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-assignment
-        helpers.setValue({ ...field.value, [key]: value });
-      };
+    (key: string, fn = (v: string): number | string => v) =>
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      helpers.setTouched(true);
+      const cast = fn(e.target.value);
+      const value =
+        e.target.type === "number"
+          ? typeof cast === "number" && isNaN(cast)
+            ? null
+            : cast
+          : cast;
+      helpers.setValue({ ...field.value, [key]: value });
+    };
 
   return (
     <Fragment>
       <FormFieldInput
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/restrict-template-expressions
         fieldName={`${prefix}.name`}
         meta={{
           ...meta,
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-assignment
-          error: meta.error && meta.error.name,
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore:disable
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+          error: meta.error && meta.error?.name,
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore:enable
         }}
         formField={{ label: `Name${requiredChar}`, placeholder: "main" }}
         field={
           prefix === "config.constraint"
-            ? { value: "main", readOnly: true }
-            : // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            ? // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore:disable
+              { value: "main", readOnly: true }
```

Comment:
You can add a `fieldProps` object and pass it to `FormFieldInput` instead. This way you can pass both `readOnly` and `type:number` (see comment below)

Reply:
found a better way :)